### PR TITLE
feat(telemetry): Introduce `DO_NO_TRACK` and `KEDRO_DISABLE_TELEMETRY`

### LINKED ISSUE #728: `kedro-telemetry`: Introduce environment variables for disabling telemetry and revise disabling methods
## Description
Part of https://github.com/kedro-org/kedro-plugins/issues/715:

Introduce two new environment variables for disabling telemetry:
- `DO_NOT_TRACK`
- `KEDRO_DISABLE_TELEMETRY`
When either of these variables is set to True or 1, telemetry should not be used. This will have the same effect as setting `consent: false` in the `.telemetry` file, but without the need to modify the file. We will only check the consent.

## Context
Detailed explanation provided by @astrojuanlu in [this comment](https://github.com/kedro-org/kedro-plugins/issues/715#issuecomment-2161181479).

When all the opt-out tasks for #715 are completed these should be the ways to disable telemetry:
1. Environment Variables:
Set the environment variables `DO_NOT_TRACK` or `KEDRO_DISABLE_TELEMETRY` to 1. Setting either of these variables will fully disable telemetry for all projects in that environment.
2. Telemetry File:
Create a `.telemetry` file in the root folder of the Kedro project and write there `consent: false`. This disables telemetry for any kedro command executed within that project's folder.
3. CLI Option when creating a new project:
When creating a new project, users can use the command `kedro new --telemetry=yes/no`. This will create a `.telemetry` file with `consent: true/false` accordingly. However, telemetry data about the execution of the kedro new command will still be sent after #729 implementation.
4. `pip uninstall kedro-telemetry`
5. CLI telemetry Disable for each command execution:
Previously, we attempted to implement a CLI consent disable for `kedro run` command (see [Issue 2867](https://github.com/kedro-org/kedro/issues/2867)), but it was reverted due to unclear initial design. Considering the shift to an opt-out model, I believe @ankatiyar's [proposal](https://github.com/kedro-org/kedro-plugins/pull/681#issuecomment-2127020217) to provide consent for each execution of certain Kedro commands is still relevant. 

**Let's discuss this proposal**:
- Implement a `--no-telemetry` flag for `kedro run` and possibly other commands: `kedro run --no-telemetry`.
- This flag will disable telemetry only for that execution of that particular command.
- This can be beneficial for users who want to run their projects in CI/CD processes without needing to set environment variables.

### PR DESCRIPTION
**Pull request by noklam**
## Description

Fix #728
## Development notes

- Added two new env vars to skip telemetry if exist. It will override the consent regardless what is written in a `.telemetry` file.
- 
## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-telemetry/RELEASE.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Upcoming release
+* Added `DO_NOT_TRACK` and `KEDOR_DISABLE_TELEMETRY` environment variables to skip telemetry.
```
Comment (lrcouto) [inline kedro-telemetry/RELEASE.md:2]:
```suggestion
* Added `DO_NOT_TRACK` and `KEDRO_DISABLE_TELEMETRY` environment variables to skip telemetry.
```

**Review by lrcouto — APPROVED** [review]
This looks good to me. I tested it locally and it works fine.

**Comment by noklam** [comment]
CI error is irrelevant, suspect some changes caused by https://github.com/kedro-org/kedro/pull/3883.

**Review by DimedS — APPROVED** [review]
Thank you, @noklam! I appreciate the flow where these environment variables take first priority only when they are set to True.

**Pull request merged**